table/block_based/block_based_table_builder.cc

```diff
@@ -461,6 +465,12 @@ struct BlockBasedTableBuilder::Rep {
       buffer_limit = std::min(tbo.target_file_size,
                               compression_opts.max_dict_buffer_bytes);
     }
+    // TODO figure out c++ member initialization syntax
+    buffer_exceeds_global_memory_limit = false;
+    cache_rep = (table_options.block_cache != nullptr
+                     ? new CacheRep(table_options.block_cache.get())
```

Comment:
Question 1:
While the block_cache is WriteBufferManager is passed in on WBM's creation, I believe we should the account memory in the block cache set in the table_options. Did I understand it correctly?

Reply:
Just realized that on https://github.com/facebook/rocksdb/wiki/Block-Cache#usage, it says "RocksDB will create the default block cache if block_cache is set to nullptr. To disable block cache completely: ...". 

So it seems that there are cases where (1)user completely disabled block cache - in this case, costing buffer memory is not needed (2)user didn't disable block cache and block_cache == nullptr, RocksDB provides a default block cache (3)user passed in its block cache. 

I will temporarily close this question until I read more codes to figure out how to include case (1), (2) in my implementation.